symengine/complex.h

```diff
@@ -64,15 +64,15 @@ class Complex : public Number {
     static RCP<const Number> from_two_nums(const Number &re,
         const Number &im);
 
-    //! \return `true` if `0`
-    virtual bool is_zero() const { return ((this->real_ == 0) and (this->imaginary_ == 0)); }
-    //! \return `true` if `1`
-    virtual bool is_one() const { return ((this->real_ == 1) and (this->imaginary_ == 0)); }
-    //! \return `true` if `-1`
-    virtual bool is_minus_one() const { return ((this->real_ == -1) and (this->imaginary_ == 0)); }
-    //! \return `true` if both `real_` or `imaginary_` are zero
+    //! \return false since imaginary cannot be zero
+    virtual bool is_zero() const { return false; }
+    //! \return false since imaginary cannot be zero
+    virtual bool is_one() const { return false; }
+    //! \return false since imaginary cannot be zero
+    virtual bool is_minus_one() const { return false; }
+    //! \return `true` if both `real_` is zero
```

Comment:
`is_reim_zero()` and `is_re_zero()` are both the same now. I would replace `is_reim_zero()` calls with `is_re_zero()`.